### PR TITLE
MLNodeABecLaplacian

### PR DESCRIPTION
**Pull request by WeiqunZhang**
Add a new Linear Opearator class MLNodeABecLaplacian.

It solves `(alpha * a - beta * (del dot b grad)) phi = rhs`, where a, phi and rhs are nodal MultiFabs and b is cell-centered.

It works in 2D and 3D, CPU and GPU. So far it works only for single level.

There is an example of using it at Tests/LinearSolvers/ABecLaplacian_C as prob_type=4. The test problem converges at the second order.

## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [x] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

**Comment by WeiqunZhang** [comment]
Ccache on windows is broken for unknown reasons. Maybe it will come back tomorrow.

File: Tests/LinearSolvers/ABecLaplacian_C/inputs-node [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+max_level = 0
+ref_ratio = 2
+n_cell = 128
+max_grid_size = 64
+
+composite_solve = 0   # composite solve or level by level?
+
+prob_type = 4 # nodal laplacian
```
Comment (ajnonaka) [inline Tests/LinearSolvers/ABecLaplacian_C/inputs-node]:
```suggestion
prob_type = 4 # nodal ABecLaplacian
```

**Review by ajnonaka — APPROVED** [review]
Approve, as long as the failing CI build tests are actually not an issue.

File: Tests/LinearSolvers/ABecLaplacian_C/MyTestPlotfile.cpp [hunk 2]
```diff
@@ -8,6 +8,19 @@ using namespace amrex;
...
 {
+    if (prob_type == 4) {
+        for (int ilev = 0; ilev <= max_level; ++ilev) {
+            VisMF::Write(solution[ilev], "solution-lev"+std::to_string(ilev));
+            MultiFab errmf(solution[ilev].boxArray(),
+                           solution[ilev].DistributionMap(), 1, 1);
+            MultiFab::Copy(errmf, solution[ilev], 0, 0, 1, 0);
+            MultiFab::Subtract(errmf, exact_solution[ilev], 0, 0, 1, 0);
+            auto error = errmf.norminf();
+            amrex::Print() << "Leve " << ilev << " max-norm error: " << error << std::endl;
```
Comment (ajnonaka) [inline Tests/LinearSolvers/ABecLaplacian_C/MyTestPlotfile.cpp]:
```suggestion
            amrex::Print() << "Level " << ilev << " max-norm error: " << error << std::endl;
```

File: Src/LinearSolvers/MLMG/AMReX_MLNodeABecLaplacian.cpp [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+
+void
+MLNodeABecLaplacian::define (const Vector<Geometry>& a_geom,
+                             const Vector<BoxArray>& a_grids,
+                             const Vector<DistributionMapping>& a_dmap,
+                             const LPInfo& a_info,
+                             const Vector<FabFactory<FArrayBox> const*>& a_factory)
+{
+#ifdef AMREX_USE_EB
+    amrex::Abort("MLNodeABecLaplacian does not suport EB");
```
Comment (ajnonaka) [inline Src/LinearSolvers/MLMG/AMReX_MLNodeABecLaplacian.cpp]:
```suggestion
    amrex::Abort("MLNodeABecLaplacian does not support EB");
```

**Comment by WeiqunZhang** [comment]
We are having Windows CI issues because ccache no longer works. There is a PR to disable it. (#3566)

The NVHPC issue might be due to disk space issue.

**Comment by WeiqunZhang** [comment]
In any case, I plan to merge this after the 23.10 release.

**Pull request merged**